chef/cookbooks/hyperv/recipes/config_neutron.rb

```diff
@@ -0,0 +1,64 @@
+raise if not node[:platform] == 'windows'
+
+keystone_settings = KeystoneHelper.keystone_settings(node, :nova)
+
+neutron_servers = search(:node, "roles:neutron-server")
+if neutron_servers.length > 0
+  neutron_server = neutron_servers[0]
+  neutron_server = node if neutron_server.name == node.name
+  neutron_protocol = neutron_server[:neutron][:api][:protocol]
+  neutron_server_host = CrowbarHelper.get_host_for_admin_url(neutron_server, (neutron_server[:neutron][:ha][:server][:enabled] rescue false))
+  neutron_server_port = neutron_server[:neutron][:api][:service_port]
+  neutron_insecure = neutron_protocol == 'https' && neutron_server[:neutron][:ssl][:insecure]
+  neutron_service_user = neutron_server[:neutron][:service_user]
+  neutron_service_password = neutron_server[:neutron][:service_password]
+  neutron_networking_plugin = neutron_server[:neutron][:networking_plugin]
+  neutron_ml2_type_drivers = neutron_server[:neutron][:ml2_type_drivers]
+else
+  neutron_server_host = nil
+  neutron_server_port = nil
+  neutron_service_user = nil
+  neutron_service_password = nil
+  neutron_networking_plugin = "ml2"
+  neutron_ml2_type_drivers = []
+end
+Chef::Log.info("Neutron server at #{neutron_server_host}")
+
+dirs = [ node[:openstack][:instances], node[:openstack][:config], node[:openstack][:bin], node[:openstack][:log] ]
+dirs.each do |dir|
+  directory dir do
+    action :create
+    recursive true
+  end
+end
+
+%w{ OpenStackService.exe }.each do |bin_file|
+  cookbook_file "#{node[:openstack][:bin]}/#{bin_file}" do
+    source bin_file
+  end
+end
```

Comment:
So the creation of the directories and of this exec is something that is done (or if not, should be done) in the three config recipes. Would it make sense to have a config_common_openstack.rb recipe for that kind of things?